src/core/surface/secure_channel_create.c

```diff
@@ -306,6 +306,13 @@ grpc_channel *grpc_secure_channel_create(grpc_channel_credentials *creds,
   GRPC_CHANNEL_INTERNAL_REF(channel, "subchannel_factory");
   resolver = grpc_resolver_create(target, &f->base);
   if (!resolver) {
+    grpc_subchannel_factory_unref(&exec_ctx, &f->base);
```

Comment:
minor: This cleanup code is also repeated below. I was wondering if perhaps you can refactor this so that this block of lines is only called once.

I mean something like  
if (resolver) {
  //  set resolver on client channel
  //  resolver unref 
}

// clean up code

// return channel if resolver is not null..